avalon/nuke/lib.py

```diff
@@ -0,0 +1,240 @@
+import os
+import contextlib
+import nuke
+import re
+import logging
+import toml
+
+log = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def maintained_selection():
+    nodes = nuke.allNodes()
+    previous_selection = [n.name()
+                          for n in nodes
+                          if n['selected'].value() is True]
+    try:
+        yield
+    finally:
+        # unselect all selection in case there is some
+        [n['selected'].setValue(False) for n in nodes]
+        # and select all previously selected nodes
+        if previous_selection:
+            [n['selected'].setValue(True)
+             for n in nodes
+             if n.name() in previous_selection]
```

Comment:
Couldn't this just be `for n in previous_selection` if you don't store the `name` but just keep a reference to the node?

Reply:
Very true